src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageAnalysisState.cs

```diff
@@ -2,73 +2,41 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading;
-using System.Threading.Tasks;
+using Microsoft.DotNet.UpgradeAssistant.Packages;
 
 namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages
 {
-    public class PackageAnalysisState
+    internal class PackageAnalysisState : IDependencyAnalysisState
     {
-        public IList<Reference> FrameworkReferencesToAdd { get; }
-
-        public IList<Reference> FrameworkReferencesToRemove { get; }
-
-        public IList<NuGetReference> PackagesToAdd { get; }
-
-        public IList<NuGetReference> PackagesToRemove { get; }
-
-        public IList<Reference> ReferencesToRemove { get; }
+        public PackageAnalysisState(IProject project, INuGetReferences nugetReferences)
+        {
+            FrameworkReferences = new PackageCollection<Reference>(project.FrameworkReferences, SetRisk);
+            Packages = new PackageCollection<NuGetReference>(nugetReferences.PackageReferences, SetRisk);
+            References = new PackageCollection<Reference>(project.References, SetRisk);
 
-        public bool Failed { get; set; }
+            void SetRisk(BuildBreakRisk risk)
```

Comment:
This will preserve our old behavior but now that you have a `BuildBreakRisk` available, we can do better. Rather than just set `PossibleBreakingChangeRecommended` (which just gets translated to 'none' or 'medium' risk by the upgrade step), we should just track the highest risk set so far and expose that to the upgrade step to use.